aptos-move/aptos-gas-meter/src/algebra.rs

```diff
@@ -40,15 +41,23 @@ pub struct StandardGasAlgebra {
 
     num_dependencies: NumModules,
     total_dependency_size: NumBytes,
+
+    // Block synchronization kill switch allows checking whether the ongoing execution should
+    // be interrupted, due to external (block execution related) conditions (such as block gas
+    // limit being reached). Interrupting is a performance optimization, and requires checking
+    // with proper granularity. Gas charging happens regularly but involves computation that
+    // can amortize the cost of the check. Hence, currently kill switch is integrated here.
+    block_synchronization_kill_switch: &'a dyn BlockSynchronizationKillSwitch,
```

Comment:
Probably we can use a compile-time generic type paramter with trait bound here in the future.

Reply:
Done